### DISCUSSION TITLE
Don't show error popup when not connected to internet

### DISCUSSION THREAD

**Discussion by verhovsky** [community/community#55526] [Copilot Conversations]
To reproduce: turn off your WiFi/network and open VS Code.

Copilot will show a popup saying how it couldn't connect to the server, with an all caps error code with underscores and all that. It's annoying.

If it's possible for VS Code extensions to tell that the computer has no network connection, then it should only show the error if it has a network connection but still can't connect to the Copilot servers. Simply changing the icon in the bottom bar to light gray or something would be sufficient. I know that I don't have internet, you don't need to remind me.

**Comment by Akash1134** [comment]
Hi @verhovsky ,

The error you're getting indicates that you either do not have a GitHub Copilot subscription, or there was an error connecting to the GitHub API to request a token to use GitHub Copilot.

To request another token from api.github.com, try signing in and out of GitHub Copilot from your IDE. Once you've logged out, GitHub Copilot will prompt you to sign back in.

Another reason this might be happening is  because the "builtin" GitHub Authentication extension in VSCode has somehow become disabled. If you search in extensions for @builtin GitHub Authentication, you should find it and you can make sure it's enabled. If it's already enabled, try disabling and re-enabling.

Let me know how it goes. Thanks!

**Reply by verhovsky** [reply]
No, I'm getting this error because I'm not connected to the internet. Copilot works fine when I have internet enabled. I'm just suggesting not showing the error when I'm not connected to the internet because I don't need VS Code to remind me that I'm  not connected to the internet whenever I start it.
